### PR TITLE
Update buttons to the new set

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
- Added the state colours required for the new buttons
- Add the replace the primary button with the new set
- Added the buttons to the demo

## QA
- Run `gulp build` and check the buttons look good in the demo

### CODE REVIEW

**Comment by richmccartney** [comment]
Hey @anthonydillon can you update the dev dependancies to this then I can run ```npm install``` on OS X with no errors:

`"devDependencies": {
        "gulp": "3.9",
        "gulp-autoprefixer": "3.1.0",
        "gulp-cache": "0.2.8",
        "gulp-cssnano": "^2.1.0",
        "gulp-notify": "2.2.0",
        "gulp-rename": "1.2.0",
        "gulp-sass": "2.1.1",
        "gulp-scss-lint": "0.1.10",
        "gulp-util": "3.0.4",
        "sassdoc": "2.1.7"
    },`

**Comment by anthonydillon** [comment]
@richmccartney done

File: scss/modules/_buttons.scss [hunk 1]
```diff
@@ -1,13 +1,180 @@
...
+  .button--inline-base {
+    display: inline-block;
+    width: auto;
+  }
+}
+
+/// Default button pattern styles
+/// @group Buttons
+%button-pattern {
+  font-smoothing: subpixel-antialiased;
```
Comment (barrymcgee) [inline scss/modules/_buttons.scss]: Shouldn't this be site wide?

**Comment by anthonydillon** [comment]
Design:
https://cloud.githubusercontent.com/assets/9105999/13431866/0f5d4906-dfc3-11e5-91f1-d55f7d7f2bcb.png

File: scss/modules/_buttons.scss (hunk 1 above)
Comment (anthonydillon) [inline scss/modules/_buttons.scss]: Yeah you are right. I'll remove from here for now.

**Comment by barrymcgee** [comment]
Aside from note, I'm seeing this warning when running the build;

`Autoprefixer's process() method is deprecated and will removed in next major release. Use postcss([autoprefixer]).process() instead`

All else LGTM

**Comment by richmccartney** [comment]
Should we make this new button style a mixin too? So you could pass the button state and reference it to a colour similar to our error states on forms? Means we could then just add as many as we need.

Also I think you could add the blue button and call it primary.

**Comment by barrymcgee** [comment]
@anthonydillon You have removed 'gulp-minify-css' from `package.json` but it's still included at the top of the `Gulpfile.js` resulting in an error;

`Error: Cannot find module 'gulp-minify-css'`

**Comment by anthonydillon** [comment]
The reason I like the declarative methodology for the buttons component, is
that it is easier to read and understand. If anyone reads our Sass they
will understand exactly which buttons are available and what they will look
and act like.

It is also self documenting. As we can describe each button is Sass docs
above each mixin.

I understand it becomes a bit more complex to add additional buttons to the
theme but I believe that is a good thing. It shouldn't be trivial to modify
our components. Resulting in a little push back from the developers when
design does not match our framework/theme.

I'm happy to go with consensus on this one but that is my opinion.

**Comment by anthonydillon** [comment]
@barrymcgee Good spot. Updated the Gulpfile.

**Comment by barrymcgee** [comment]
Now i'm seeing; `ReferenceError: sourcemaps is not defined`

Think you need to include: `var sourcemaps = require('gulp-sourcemaps');`

**Comment by richmccartney** [comment]
About to say the same thing @barrymcgee :P

**Comment by anthonydillon** [comment]
Removed sorry about this.

**Comment by richmccartney** [comment]
if we were to change the class names to go ```.button--destructive``` and ```.button--destructive-inline``` we could put the ```inline``` class extension into the pattern and reduce the amount of repetitive CSS.

**Comment by richmccartney** [comment]
Also would fit the BEM mantra a bit more I feel :)

**Comment by barrymcgee** [comment]
@ant Now builds without error :)

**Comment by richmccartney** [comment]
One more thing these buttons don't overwrite a few styles if the class is applied to an ```<a href="#">``` or a ```<input type="submit">```

**Comment by richmccartney** [comment]
+1

**Comment by anthonydillon** [comment]
After a discussion with @richmccartney we'll go with what we have as you cant expend in a placeholder.

**Pull request merged**